cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/CompositeDatasetAdmin.java

```diff
@@ -18,88 +18,92 @@
 
 import co.cask.cdap.api.annotation.Beta;
 import co.cask.cdap.api.dataset.DatasetAdmin;
+import co.cask.cdap.api.dataset.DatasetSpecification;
+import co.cask.cdap.api.dataset.Updatable;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
 import java.util.Collections;
-import java.util.List;
+import java.util.Map;
 
 /**
  * Implementation of {@link DatasetAdmin} that propagates administrative operations to the given list of
  * {@link co.cask.cdap.api.dataset.DatasetAdmin}s
  */
 @Beta
-public class CompositeDatasetAdmin implements DatasetAdmin {
+public class CompositeDatasetAdmin implements DatasetAdmin, Updatable {
 
   private static final Logger LOG = LoggerFactory.getLogger(CompositeDatasetAdmin.class);
 
-  private final List<DatasetAdmin> delegates;
+  protected final Map<String, DatasetAdmin> delegates;
 
   /**
    * Constructor that takes list of dataset admins
    * @param admins list of dataset admins
    */
-  public CompositeDatasetAdmin(Collection<? extends DatasetAdmin> admins) {
-    this.delegates = Collections.unmodifiableList(new ArrayList<>(admins));
-  }
-
-  /**
-   * Constructor that takes list of dataset admins
-   * @param admins list of dataset admins
-   */
-  public CompositeDatasetAdmin(DatasetAdmin... admins) {
-    this(Arrays.asList(admins));
+  public CompositeDatasetAdmin(Map<String, ? extends DatasetAdmin> admins) {
+    if (admins == null) {
+      throw new IllegalArgumentException("delegates map must not be null");
+    }
+    this.delegates = Collections.unmodifiableMap(admins);
```

Comment:
hmmm in the way that it is being used it should not be necessary, but it does not harm. Will do.